docs/sources/mimir/manage/tools/mimirtool.md

```diff
@@ -163,11 +163,40 @@ The following command deletes the Alertmanager configuration in the Grafana Mimi
 mimirtool alertmanager delete
 ```
 
-#### Migrate Alertmanager configuration for UTF-8 in Alertmanager 0.27 and later
+#### Migrate Alertmanager configuration for UTF-8 in Mimir 2.12 and newer
 
-The migrate-utf8 command translates any matchers that are incompatible with the UTF-8 parser into
-equivalent matchers that are compatible by enforcing double quoting on the right hand side
-of the matcher. The translation does not change their behavior.
+This requires mimirtool version 2.12 or newer. You can check the version of mimirtool with
+`mimirtool version`.
+
+In accordance with [prometheus/prometheus#13095](https://github.com/prometheus/prometheus/issues/13095)
+and [prometheus/alertmanager#3486](https://github.com/prometheus/alertmanager/issues/3486), Mimir is
+adding support for UTF-8. To support UTF-8 in alerts, routes, silences and inhibition rules,
+Alertmanager has added a new parser for matchers that has a number of backwards incompatible changes.
+More information about these changes can be found [here](https://prometheus.io/docs/alerting/latest/configuration/#label-matchers).
+
+The migrate-utf8 command migrates an existing Alertmanager configuration in preparation for when
+UTF-8 is enabled in a Mimir installation. To do this, it translates matchers that are incompatible
+with UTF-8 into equivalent matchers that are compatible. This translation is backwards compatible
+and does not change the behavior of existing matchers, and works even in Mimir installations that
+do not have UTF-8 enabled.
+
+A lot of Alertmanager configurations will not need migrating. You can verify if an existing
+Alertmanager configuration needs to be migrated for UTF-8 by running the verify command:
+
+```bash
+mimirtool alertmanager verify <config_file> [template_files...]
+```
+
+If the command succeeds without any warnings, the configuration is compatible with UTF-8 and no
+migration is required.
+
+However, if the command prints the following warning(s):
+
+```
+Alertmanager is moving to a new parser for labels and matchers, and this input is incompatible. Alertmanager has instead parsed the input using the classic matchers parser as a fallback. To make this input compatible with the UTF-8 matchers parser please make sure all regular expressions and values are double-quoted. If you are still seeing this message please open an issue.
+```
+
+then the Alertmanager configuration needs to be migrated with the migrate-utf8 command.
```

Comment:
```suggestion
However, if the command prints the following warning, then you need to migrate the Alertmanager configuration with the migrate-utf8 command.

```
Alertmanager is moving to a new parser for labels and matchers, and this input is incompatible. Alertmanager has instead parsed the input using the classic matchers parser as a fallback. To make this input compatible with the UTF-8 matchers parser, ensure that all regular expressions and values are double-quoted. If you still see this message, open an issue.
```


```

Reply:
We try to avoid using "please" in documentation and error messages.

Reply:
I think it reads a bit better if we show the error message below the conditional sentence.

Reply:
Ah. I can't remove the `please` as this error message comes from Prometheus. But I made all other changes.